apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java

```diff
@@ -381,81 +424,95 @@ public Uni<Boolean> authorizeInternal(StargateBridge bridge, Schema.SchemaRead s
   }
 
   // gets a keyspace by provided name
+  // if validate hash is false, cached keyspaces are not validated to have correct hash
   private Uni<Schema.CqlKeyspaceDescribe> getKeyspaceInternal(
-      StargateBridge bridge, String keyspaceName) {
+      StargateBridge bridge, String keyspaceName, boolean validateHash) {
     Optional<String> tenantId = requestInfo.getTenantId();
 
-    // check if cached, if so we need to revalidate hash
+    // check if cached
     return Uni.createFrom()
         .deferred(
             () -> {
               CompositeCacheKey cacheKey = new CompositeCacheKey(keyspaceName, tenantId);
-              CompletableFuture<Object> keyspace =
+              CompletableFuture<Object> keyspaceFuture =
                   keyspaceCache.as(CaffeineCache.class).getIfPresent(cacheKey);
 
-              // if exists ensure it's not null
-              if (null != keyspace) {
-                return Uni.createFrom().future(keyspace).map(Objects::nonNull);
+              // keyspace might be null
+              if (null != keyspaceFuture) {
+                // if it's not null, we map to keyspace with cached flag true
+                return Uni.createFrom()
+                    .future(keyspaceFuture)
+                    .map(
+                        k -> {
+                          // we don't know what's the cached object
+                          // errors are also cached, so we need to add explicit check
+                          if (k instanceof Schema.CqlKeyspaceDescribe cqlKeyspace) {
+                            return Tuple2.of(cqlKeyspace, true);
+                          }
+                          return null;
+                        });
               } else {
-                return Uni.createFrom().item(false);
+                // otherwise push null uni down
+                return Uni.createFrom().nullItem();
               }
             })
+
+        // if it was not cached in first place, fetch from the keyspace from bridge
+        // and return with cached false
+        .onItem()
+        .ifNull()
+        .switchTo(() -> fetchKeyspace(keyspaceName, tenantId, bridge).map(k -> Tuple2.of(k, false)))
```

Comment:
Good -- I much prefer this version to the earlier version! Much easier to read, follow what is happening. Plus avoids extra cache access too,